File: cpp_solutions/1342. Number of Steps to Reduce a Number to Zero.cpp
class Solution {
public:
    int numberOfSteps(int num) {
        
        int steps = 0;
        
        while(num!=0){
            steps+=1;
            if (num%2==0){
                num = num/2;
            }
            else{
                num-=1;
            }
        }
        return steps;
        
    }
};

File: python3_solution_set0/1342. Number of Steps to Reduce a Number to Zero.py
class Solution:
    def numberOfSteps (self, num: int) -> int:
        
        count = 0
        while num !=0:
            
            if(num%2==0):
                num = num/2
                count = count + 1
            
            else:
                num = num -1
                count = count +1
            
        return count
        
